File: FMELibrary/Kit.cs
namespace FMMLibrary
{
    /// <summary>
    /// Represents a football kit (uniform) with its colors and properties.
    /// </summary>
    public class Kit
    {
        /// <summary>
        /// Gets or sets an unknown byte value.
        /// </summary>
        public byte Unknown1 { get; set; }

        /// <summary>
        /// Gets or sets an unknown byte value.
        /// </summary>
        public byte Unknown2 { get; set; }

        /// <summary>
        /// Gets or sets the array of 10 color values for the kit.
        /// </summary>
        public short[] Colors { get; set; }

        /// <summary>
        /// Initializes a new instance of the <see cref="Kit"/> class by reading from a binary reader.
        /// </summary>
        /// <param name="reader">The binary reader containing the kit data.</param>
        public Kit(BinaryReader reader)
        {
            Unknown1 = reader.ReadByte();
            Unknown2 = reader.ReadByte();

            Colors = new short[10];
            for (int i = 0; i < Colors.Length; i++)
            {
                Colors[i] = reader.ReadInt16();
            }
        }

        /// <summary>
        /// Writes the kit data to the specified binary writer.
        /// </summary>
        /// <param name="writer">The binary writer to write the kit data to.</param>
        public void Write(BinaryWriter writer)
        {
            writer.Write(Unknown1);
            writer.Write(Unknown2);

            for (int i = 0; i < Colors.Length; i++)
                writer.Write(Colors[i]);
        }
    }
}

File: FMELibrary.js/Kit.ts
/**
 * Represents a football kit (uniform) with its colors and properties.
 */
export class Kit {
  /**
   * Gets or sets an unknown byte value.
   */
  unknown1: number = 0;

  /**
     * Gets or sets an unknown byte value.
     */
  unknown2: number = 0;

  /**
   * Gets or sets the array of 10 color values for the kit.
   */
  colors: number[] = [];

  /**
   * Initializes a new instance of the Kit class by reading from a buffer.
   * @param buffer - The buffer containing the kit data
   * @param offset - The current offset in the buffer
   * @returns An object containing the Kit instance and the new offset
   */
  static fromBuffer(buffer: Buffer, offset: number): { kit: Kit; offset: number } {
    const kit = new Kit();

    kit.unknown1 = buffer.readUInt8(offset);
    offset += 1;

    kit.unknown2 = buffer.readUInt8(offset);
    offset += 1;

    kit.colors = [];
    for (let i = 0; i < 10; i++) {
      kit.colors.push(buffer.readInt16LE(offset));
      offset += 2;
    }

    return { kit, offset };
  }

  /**
   * Writes the kit data to the specified buffer.
   * @param buffer - The buffer to write the kit data to
 * @param offset - The current offset in the buffer
   * @returns The new offset after writing
   */
  write(buffer: Buffer, offset: number): number {
    buffer.writeUInt8(this.unknown1, offset);
    offset += 1;

    buffer.writeUInt8(this.unknown2, offset);
    offset += 1;

    for (let i = 0; i < this.colors.length; i++) {
      buffer.writeInt16LE(this.colors[i], offset);
      offset += 2;
    }

    return offset;
  }

  /**
   * Calculates the size of the kit data in bytes.
   * @returns The size in bytes (22 bytes: 1 + 1 + 10*2)
   */
  static getSize(): number {
    return 1 + 1 + (10 * 2); // unknown1 + unknown2 + 10 colors
  }
}
